src/data/fun_like/equiv.lean

```diff
@@ -117,34 +118,37 @@ instead of linearly increasing the work per `my_iso`-related declaration.
 
 -/
 
+set_option old_structure_cmd true
+
 /-- The class `equiv_like E α β` expresses that terms of type `E` have an
 injective coercion to bijections between `α` and `β`.
 
-This typeclass is used in the definition of the homomorphism typeclasses,
+This typeclass is used in the definition of the isomorphism typeclasses,
 such as `zero_equiv_class`, `mul_equiv_class`, `monoid_equiv_class`, ....
 -/
-class equiv_like (E : Sort*) (α β : out_param Sort*) :=
-(coe : E → α → β)
+class equiv_like (E : Sort*) (α β : out_param Sort*) extends fun_like E α (λ _, β) :=
 (inv : E → β → α)
-(left_inv  : ∀ e, function.left_inverse (inv e) (coe e))
-(right_inv : ∀ e, function.right_inverse (inv e) (coe e))
--- The `inv` hypothesis makes this easier to prove with `congr'`
-(coe_injective' : ∀ e g, coe e = coe g → inv e = inv g → e = g)
+-- the `show` below fixes elaboration problems when using these fields stemming from the fact
+-- that in this case, `fun_like.coe : E → Π (a : α), (λ _, β) a`, not `E → α → β`.
+(left_inv  : ∀ e, function.left_inverse (inv e) (show α → β, from coe e))
+(right_inv : ∀ e, function.right_inverse (inv e) (show α → β, from coe e))
+(coe_inv_injective' : ∀ e g, coe e = coe g → inv e = inv g → e = g)
```

Comment:
I'm not totally convinced by the name since I intuitively parse it like `function.injective ↑(equiv_like.inv _)`. I can't think of anything better so this is probably fine.